Don't Infer Number of AQUFETP Records from AQUDIMS

bska opened the pull request:
Original (and incorrect) description:

Without this you get
```sh
$ opmpack path/to/opm-tests/flow_diagnostic_test/SIMPLE_SUMMARY2.DATA > /dev/null
terminate called after throwing an instance of 'std::invalid_argument'
  what():  
Failed to parse keyword: AQUCT
Starting at location: path/to/opm-tests/flow_diagnostic_test/SIMPLE_SUMMARY2.DATA(189)

Inner exception: Malformed integer 'AQUANCON'

Aborted
```
This PR makes the `opmpack` command succeed.

---

Updated description:

The number of records is unrelated to AQUDIMS Item 5 (NANAQU).  That item sets the maximum aquifer ID permissible as Item 1 of AQUFETP (and other keywords related to analytic aquifers).

Update unit test accordingly.

bska commented:
jenkins build this please

joakim-hove commented:
Hmmmm; I dont' really understand the issue here - but the fix does not seem correct to me - at least the `AQUCT` documentation ends with the sentence: 
```
   The set of records should be terminated with a blank record containing just a slash.
```
So I would maybe argue that this looks like user error (i.e. missing `/`) in the deck in question? On th other hand that the initial part of the documentation indicates that your fix is correct. I'll let you decide

bska commented:
> Hmmmm; I dont' really understand the issue here - but the fix does not seem correct to me

You're right&mdash;this PR is incorrect (and `AQUFETP` has the same problem).  The `NANAQU` item of `AQUDIMS` does not refer to the number of records.  Rather, it refers to the maximum aquifer ID permissible as Item 1 of `AQUCT` (and others).  I'll revert this and update the `AQFETP` description instead.

bska commented:
I have updated the PR (and changed its title) to remove the `size` parameter from the description of `AQUFETP` instead of the original submission.

It is a little disconcerting though that ECLIPSE appears to be able to surmise the intention of an `AQUCT` specification that is not terminated by an empty record.  @joakim-hove: Would it be possible to support that mode of operation in our current system?  If so, how much work would that involve?

bska commented:
jenkins build this please

joakim-hove commented:
> Would it be possible to support that mode of operation in our current system? If so, how much work would that involve?

Looking at the documentation of the `AQUCT` keyword it seems that both slash termination, as the current version of the PR does - and getting the number of records from the `AQUDIMS` record could be correct interpretations.  Assuming that this is indeed the case we could change parsing for the situation where the the number of records is given by an item in another keyword with one or both of these changes:

1. If there is an extra terminating `/` at the end of keyword - which already has the correct number of records given by a `xxxDIMS` item - we silently ignore the `/` - i.e. we do not even flag a `PARSECONTEXT_RANDOM_SLASH` error.

2. If a keyword with a fixed number of records is prematurely terminated with a `/` we accept that, without screaming of missing records. 

These two items are independent; the second certainly has the most significant possible problems. Both of these should be quite doable tasks (two working days?) - personally I think 1) is a good idea, more skeptical to 2). 

If we embark on this it will of course be general for all keywords where the number of records is given from a `xxxDIMS` item, special treatment of `AQUCT` will of course break our model severely, and that I will really oppose.

OPMUSER commented:
Just for clarification, like nearly all dimension statement keywords in the RUNSPEC section AQUDIMS sets the maximum number of entries Remember the commercial simulator is FORTRAN based not C++. So if the number of entries exceed the dimension specification then that should result in an ERROR/WARNING message.

Apologies for the late reply.
